### PR TITLE
Clone private engines at the right tag

### PR DESCRIPTION
**Pull request by kirillseva**
Tested locally. It now clones source at a given tag with `depth = 1` (meaning no useless history in your engine cache) and does so quietly, while respecting your github_pat

### CODE REVIEW

File: R/engine.R [hunk 1]
```diff
@@ -412,11 +412,18 @@ parse_engine.github <- function(engine_parameters) {
   repo    <- engine_parameters$repo %||% engine_parameters$repository
   # TODO: (RK) Checking for updates?
   version <- engine_parameters$version %||% "master"
+  base_url <- if (nzchar(PAT <- github_pat())) {
```
Comment (robertzk) [inline R/engine.R]: Can you move `PAT <- github_pat()` to the line above?

Comment (robertzk) [inline R/engine.R]: Also be careful of 

```r
> nzchar(NULL)
logical(0)
```

File: R/utils.R [hunk 2]
```diff
@@ -7,11 +7,28 @@
...
 `%|||%`  <- function(x, y) if (is.falsy(x)) y else x
+
 is.falsy <- function(x) {
   identical(x, NULL) || identical(x, FALSE) || identical(x, "") ||
   length(x) == 0 || identical(x, 0)
 }
 
+#' Retrieve Github personal access token.
+#'
+#' Borrowed from https://github.com/r-pkgs/remotes/blob/master/R/github.R#L23
```
Comment (robertzk) [inline R/utils.R]: Use `\url{}`

File: R/utils.R [hunk 3]
```diff
@@ -7,11 +7,28 @@
...
+#' A github personal access token
+#' Looks in env var \code{GITHUB_PAT}
+#'
+#' @keywords internal
+#' @noRd
+github_pat <- function() {
+  pat <- Sys.getenv('GITHUB_PAT')
+  if (identical(pat, "")) { return(NULL) }
```
Comment (robertzk) [inline R/utils.R]: `nzchar`

Comment (kirillseva) [inline R/utils.R]: this one is copypasterino, but I'll fix

**Pull request merged**